pkg/build/util/util.go

```diff
@@ -114,25 +113,23 @@ func BuildConfigSelectorDeprecated(name string) labels.Selector {
 	return labels.Set{buildapi.BuildConfigLabelDeprecated: name}.AsSelector()
 }
 
-type buildFilter func(buildapi.Build) bool
+type buildFilter func(*buildapi.Build) bool
 
 // BuildConfigBuilds return a list of builds for the given build config.
 // Optionally you can specify a filter function to select only builds that
 // matches your criteria.
-func BuildConfigBuilds(c buildclient.BuildLister, namespace, name string, filterFunc buildFilter) (*buildapi.BuildList, error) {
-	result, err := c.List(namespace, metav1.ListOptions{
-		LabelSelector: BuildConfigSelector(name).String(),
-	})
+func BuildConfigBuilds(c buildlister.BuildLister, namespace, name string, filterFunc buildFilter) ([]*buildapi.Build, error) {
+	result, err := c.Builds(namespace).List(BuildConfigSelector(name))
```

Comment:
Yeah, I meant should we introduce a new index to index builds by buildconfig key?